docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -10,7 +10,20 @@ to isolate authentication test actions and apply them easily whenever you need t
 
 A piece of logic that logs in a particular user is called a *role*. Define a role for each user account participating in your test.
 
-## Creating and Using Roles
+## Why Use Roles
+
+Roles are more than just another way to [extract reusable test logic](../../recipes/extract-reusable-test-code/README.md). They were specially designed for login operations and provide the following dedicated features:
+
+* **Object-based API.** Authentication data and logic are stored in an object that is easy to pass around and activate when needed.
+* **Single login.** Login actions are not repeated when you switch to a previously used role within the same session. If you activate a role in the [beforeEach](../test-code-structure.md#test-hooks) hook, login actions run once before the first test. The subsequent tests reuse authentication data so that it happens instantly.
+* **Automatic return.** The browser automatically navigates back to the page that was opened when you switched the role (you can disable this behavior if it doesn't suit your scenario).
```

Comment:
Should `you can disable this behavior if it doesn't suit your scenario` phrase have a link to corresponding option?

Reply:
I'm not sure it matters at this point... I'll add a link to the description in the middle.